### PR TITLE
SSB4: Frostyicelad

### PR DESCRIPTION
**Pull request by Volco**
only things that bug me personally, mega having no sprite and no ability in tooltips

### CODE REVIEW

File: data/mods/ssb/statuses.ts [hunk 1]
```diff
@@ -57,6 +57,22 @@ export const BattleStatuses: {[k: string]: ModdedPureEffectData} = {
 			}
 		},
 	},
+
+	frostyicelad: {
+		noCopy: true,
+		onStart(source) {
+			this.add(`c|${getName('Frostyicelad')}|Oh i guess its my turn now! Time to sweep!`);
+			if (source.species.id !== 'absolmega' || source.illusion) return;
```
Comment (ghost) [inline data/mods/ssb/statuses.ts]: absol

Comment (Volco) [reply data/mods/ssb/statuses.ts]: oops ever forget to edit the c/p when trying to be lazy?

File: data/mods/ssb/moves.ts [hunk 2]
```diff
@@ -76,6 +76,31 @@ export const BattleMovedex: {[k: string]: ModdedMoveData} = {
 		target: "normal",
 		type: "Fire",
 	},
+
+	frostywave: {
```
Comment (KrisXV) [inline data/mods/ssb/moves.ts:210]:
```suggestion
	// frostyicelad
	frostywave: {
```

File: data/mods/ssb/moves.ts [hunk 3]
```diff
@@ -76,6 +76,31 @@ export const BattleMovedex: {[k: string]: ModdedMoveData} = {
...
+		accuracy: 100,
+		basePower: 95,
+		category: "Special",
+		desc: "Sound based move, ignores foes abilities, targets all opponents on the field.",
```
Comment (KrisXV) [inline data/mods/ssb/moves.ts]:
```suggestion
		desc: "This move and its effects ignore the Abilities of other Pokemon.",
```

File: data/mods/ssb/moves.ts [hunk 4]
```diff
@@ -76,6 +76,31 @@ export const BattleMovedex: {[k: string]: ModdedMoveData} = {
...
+		shortDesc: "Sound Based, ignores abilities, hits all foes.",
```
Comment (KrisXV) [inline data/mods/ssb/moves.ts]:
```suggestion
		shortDesc: "Ignores abilities. Hits adjacent Pokemon.",
```

File: data/mods/ssb/random-teams.ts [hunk 5]
```diff
@@ -40,6 +40,14 @@ export const ssbSets: SSBSets = {
 		signatureMove: 'Never Lucky',
 		evs: {hp: 248, def: 36, spe: 224}, ivs: {atk: 0}, nature: 'Timid',
 	},
+
+	Frostyicelad: {
+		species: 'Frosmoth', ability: 'Ice Shield', item: 'Ice Stone', gender: 'M',
+		moves: ['Quiver Dance', 'Bug Buzz', ['Earth Power', 'Sparkling Aria']],
+		signatureMove: 'Frosty Wave',
+		evs: {spa: 252, spd: 4, spe: 252}, ivs: {atk: 0}, nature: 'Timid', level: 100, shiny: false,
```
Comment (KrisXV) [inline data/mods/ssb/random-teams.ts]:
```suggestion
		evs: {spa: 252, spd: 4, spe: 252}, ivs: {atk: 0}, nature: 'Timid',
```

File: data/mods/ssb/scripts.ts [hunk 6]
```diff
@@ -24,6 +24,7 @@ export const BattleScripts: ModdedBattleScriptsData = {
 
 		// Overneat gains the fairy type when mega evolving
 		if (pokemon.name === 'Overneat' && !pokemon.illusion) this.add('-start', pokemon, 'typeadd', 'Fairy');
+		if (pokemon.name === 'Frostyicelad' && !pokemon.illusion) this.add('-start', pokemon, 'typechange', 'Bug/Ice');
```
Comment (KrisXV) [inline data/mods/ssb/scripts.ts]: I don't think this line is necessary. It's the same type as the base forme.

File: data/mods/ssb/statuses.ts [hunk 7]
```diff
@@ -57,6 +57,22 @@ export const BattleStatuses: {[k: string]: ModdedPureEffectData} = {
 			}
 		},
 	},
+
+	frostyicelad: {
+		noCopy: true,
+		onStart(source) {
+			this.add(`c|${getName('Frostyicelad')}|Oh i guess its my turn now! Time to sweep!`);
+			if (source.species.id !== 'frosmothmega' || source.illusion) return;
+			this.add('-start', source, 'typechange', 'Bug/Ice');
```
Comment (KrisXV) [inline data/mods/ssb/statuses.ts:103]: again not necessary imo

File: data/mods/ssb/statuses.ts [hunk 8]
```diff
@@ -57,6 +57,22 @@ export const BattleStatuses: {[k: string]: ModdedPureEffectData} = {
 			}
 		},
 	},
+
+	frostyicelad: {
+		noCopy: true,
+		onStart(source) {
+			this.add(`c|${getName('Frostyicelad')}|Oh i guess its my turn now! Time to sweep!`);
```
Comment (KrisXV) [inline data/mods/ssb/statuses.ts]:
```suggestion
			this.add(`c|${getName('frostyicelad ❆')}|Oh i guess its my turn now! Time to sweep!`);
```

File: data/mods/ssb/statuses.ts [hunk 9]
```diff
@@ -57,6 +57,22 @@ export const BattleStatuses: {[k: string]: ModdedPureEffectData} = {
...
+			if (source.species.id !== 'frosmothmega' || source.illusion) return;
+			this.add('-start', source, 'typechange', 'Bug/Ice');
+		},
+		onSwitchOut(source) {
+			this.add(`c|${getName('Frostyicelad')}|Hey! ${source.side.name} why dont you keep me in and let me sweep? Mean.`);
+		},
+		onFaint() {
+			this.add(`c|${getName('Frostyicelad')}|So c-c-cold`);
```
Comment (KrisXV) [inline data/mods/ssb/statuses.ts]:
```suggestion
			this.add(`c|${getName('frostyicelad ❆')}|Hey! ${source.side.name} why dont you keep me in and let me sweep? Mean.`);
		},
		onFaint() {
			this.add(`c|${getName('frostyicelad ❆')}|So c-c-cold`);
```

**Review by KrisXV — CHANGES_REQUESTED** [review]

File: data/mods/ssb/statuses.ts (hunk 7 above)
Comment (Volco) [reply data/mods/ssb/statuses.ts:103]: until tooltips has it i view it as necessary personally, only reason i added it. if this was released to main without tooltip support for frosmoth-mega there will be waaaaaaaaaaaaaaay too many idiots being like "wHaT tYpE iS iT"

File: data/mods/ssb/scripts.ts (hunk 6 above)
Comment (Volco) [reply data/mods/ssb/scripts.ts]: until tooltips has it i view it as necessary personally, only reason i added it. if this was released to main without tooltip support for frosmoth-mega there will be waaaaaaaaaaaaaaay too many idiots being like "wHaT tYpE iS iT"

File: data/mods/ssb/statuses.ts (hunk 8 above)
Comment (Volco) [reply data/mods/ssb/statuses.ts]: i went by what frosty put on the forums
``Name: Frostyicelad``

Comment (Volco) [reply data/mods/ssb/statuses.ts]: spoke with them on main

**Pull request merged**